### PR TITLE
Filter plugin fixes

### PR DESCRIPTION
**Pull request by dwighthubbard**
This is to address issue #28 

The problem is the configuration object being used by the plugins does not load the configuration file passed as a command line argument in main() as a result the plugins are ignoring setting values from the actual configuration file.

This fixes the issue by populating the configuration singleton with the configuration file name.

This also includes some code that evidently didn't get included in the release plugin that was needed to actually validate the release versions.

### CODE REVIEW

File: src/bandersnatch_filter_plugins/blacklist_name.py [hunk 1]
```diff
@@ -31,20 +35,15 @@ def _determine_filtered_package_names(self):
         filtered_packages = set()
         try:
             lines = self.configuration['blacklist']['packages']
-            logger.debug('blacklist->packages: %r', lines)
             package_lines = lines.split('\n')
         except KeyError:
             package_lines = []
         for package_line in package_lines:
```
Comment (cooperlees) [inline src/bandersnatch_filter_plugins/blacklist_name.py]: NIT newline here is nice so it looks seperate to the try/except - Won't block on it.

File: src/bandersnatch_filter_plugins/blacklist_name.py [hunk 2]
```diff
@@ -72,26 +71,55 @@ def check_match(self, **kwargs):
...
 
-        logger.info(
-            'Checking for package %s in %r', name, self.blacklist_package_names
-        )
         if name in self.blacklist_package_names:
-            logger.debug(
-                'MATCH: Package %r is in %r', name,
-                self.blacklist_package_names
-            )
+            logger.debug(f'MATCH: Package {name!r} is in the blacklist')
```
Comment (cooperlees) [inline src/bandersnatch_filter_plugins/blacklist_name.py]: Isn't name a str, why do you want the __repr__ of it?

File: src/bandersnatch_filter_plugins/blacklist_name.py [hunk 3]
```diff
@@ -72,26 +71,55 @@ def check_match(self, **kwargs):
...
     name = 'blacklist_release'
 
+    def initialize_plugin(self):
+        """
+        Initialize the plugin
+        """
+        # Generate a list of blacklisted packages from the configuration and
+        # store it into self.blacklist_package_names attribute so this
+        # operation doesn't end up in the fastpath.
+        self.blacklist_release_requirements = \
```
Comment (cooperlees) [inline src/bandersnatch_filter_plugins/blacklist_name.py]: I prefer using () over \ for line continuation - so does the `black` code style which I will soon be converting this code base too and adding a black run into travis

File: src/bandersnatch_filter_plugins/blacklist_name.py [hunk 4]
```diff
@@ -72,26 +71,55 @@ def check_match(self, **kwargs):
...
+        try:
+            lines = self.configuration['blacklist']['packages']
+            package_lines = lines.split('\n')
+        except KeyError:
+            package_lines = []
+        for package_line in package_lines:
+            package_line = package_line.strip()
+            if not package_line or package_line.startswith('#'):
+                continue
+            filtered_requirements.add(Requirement(package_line))
```
Comment (cooperlees) [inline src/bandersnatch_filter_plugins/blacklist_name.py]: This parsing code looks like code above - utility function?

File: src/bandersnatch_filter_plugins/blacklist_name.py [hunk 5]
```diff
@@ -111,9 +139,25 @@ def check_match(self, **kwargs):
             True if it matches, False otherwise.
         """
         name = kwargs.get('name', None)
-        version = kwargs.get('version', None)
+        version_string = kwargs.get('version', None)
 
-        if not name or not version:
+        if not name or not version_string:
             return False
```
Comment (cooperlees) [inline src/bandersnatch_filter_plugins/blacklist_name.py]: maybe add a debug log for future use?

File: src/bandersnatch_filter_plugins/blacklist_name.py (hunk 2 above)
Comment (dwighthubbard) [reply src/bandersnatch_filter_plugins/blacklist_name.py]: It put's the string quotes around the name

File: src/bandersnatch_filter_plugins/blacklist_name.py (hunk 3 above)
Comment (dwighthubbard) [reply src/bandersnatch_filter_plugins/blacklist_name.py]: Unfortunetly, the ( doesn't occur in the line then until after 80 columns.

**Pull request merged**